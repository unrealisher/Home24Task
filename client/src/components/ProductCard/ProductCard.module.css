.productCard {
    display: flex;
    flex-direction: column;
    gap: 16px;
    padding: 14px 10px;
    border: 1px solid lavenderblush;
}

.button {
    padding: 0.2em;
    font-size: 16px;
    background-color: lightgoldenrodyellow;
    border: 1px solid lightgray;
    text-align: center;
    cursor: pointer;
}
