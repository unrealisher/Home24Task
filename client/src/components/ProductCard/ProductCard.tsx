import React, {useMemo} from 'react';
import { formatter } from '../../helpers/formatter';
import { IArticle } from '../../entities/article';
import styles from './ProductCard.module.css';

export interface IProductCardProps {
    article: IArticle;
}

export const ProductCard =  (props: IProductCardProps) => {
    const {article} = props;

    const price = useMemo(
        () => formatter.format(article.prices.regular.value / 100),
        [article]
    );

    return (
        <div className={styles.productCard}>
            {/*Pictures are empty, so i use others*/}
            <img src={`https://via.placeholder.com/200.png?text=${article.name}`} alt={`${article.name}`}/>
            <div>{article.name}</div>
            <div>{price}</div>
            <button className={styles.button}>Add to cart</button>
        </div>
    );
}
