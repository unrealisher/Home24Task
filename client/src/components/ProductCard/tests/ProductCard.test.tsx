import React from 'react';
import { ProductCard } from '../ProductCard';
import renderer from 'react-test-renderer';

const article = {
    name: "Polsterstuhl Aledas II",
    variantName: "Webstoff Cors: Mintgrau - 4er Set",
    prices: {
        currency: "EUR",
        regular: {
            value: 29999
        }
    },
    images: [
        {
            path: "https://cdn1.home24.net/images/media/catalog/product/200x200/png/-/1/-1000282500-210826-06123400004-IMAGE-P000000001000282500.webp"
        }
    ]
}

test('renders the ProductCard', () => {
    const tree = renderer.create(
        <ProductCard article={article} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
