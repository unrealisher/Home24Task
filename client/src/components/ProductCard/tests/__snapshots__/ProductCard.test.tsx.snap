// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ProductCard 1`] = `
<div
  className="productCard"
>
  <img
    src="https://via.placeholder.com/200.png?text=Polsterstuhl Aledas II"
  />
  <div>
    Polsterstuhl Aledas II
  </div>
  <div>
    299,99 €
  </div>
  <button
    className="button"
  >
    Add to cart
  </button>
</div>
`;
